### PR TITLE
Adds validation for connection name

### PR DESCRIPTION
**Pull request by antony-britto**

### CODE REVIEW

File: app/cdap/components/Connections/Create/index.tsx [hunk 1]
```diff
@@ -135,6 +135,9 @@ export function CreateConnection({
     if (name.trim() === '') {
       setError('Connection name must not be empty.');
       return;
+    } else if (!new RegExp('^[\\w\\s-]+$').test(name)) {
+      setError('Connection name must not contain special characters.');
```
Comment (njbriggs) [inline app/cdap/components/Connections/Create/index.tsx]: The message should be prescriptive about what is allowed (`Connection names can only include letters, numbers, and underscore`, etc.)

Comment (antony-britto) [reply app/cdap/components/Connections/Create/index.tsx]: Done

File: app/cdap/components/Connections/Create/index.tsx [hunk 2]
```diff
@@ -135,6 +135,9 @@ export function CreateConnection({
     if (name.trim() === '') {
       setError('Connection name must not be empty.');
       return;
+    } else if (!new RegExp('^[\\w\\s-]+$').test(name)) {
+      setError('Connection name can only include alphabets, numbers, hypen and underscore.');
```
Comment (njbriggs) [inline app/cdap/components/Connections/Create/index.tsx]: Nit: `letters` instead of `alphabets`

**Review by njbriggs — APPROVED** [review]
LGTM, just a slight tweak to the message

**Pull request merged**